test/e2e/basic/custom_node_labels.go

```diff
@@ -0,0 +1,78 @@
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	coreapi "k8s.io/api/core/v1"
+
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+// Test the application (and rollback) of a custom profile via node labelling.
+var _ = ginkgo.Describe("[basic][custom_node_labels] Node Tuning Operator custom profile, node labels", func() {
+	const (
+		profileHugepages   = "../../../examples/hugepages.yaml"
+		nodeLabelHugepages = "tuned.openshift.io/hugepages"
+		sysctlVar          = "vm.nr_hugepages"
+	)
+
+	ginkgo.Context("custom profile: node labels", func() {
+		var (
+			node *coreapi.Node
+		)
+
+		// Cleanup code to roll back cluster changes done by this test even if it fails in the middle of ginkgo.It()
+		ginkgo.AfterEach(func() {
+			ginkgo.By("cluster changes rollback")
+			if node != nil {
+				exec.Command("oc", "label", "node", "--overwrite", node.Name, nodeLabelHugepages+"-")
+			}
+			exec.Command("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "-f", profileHugepages)
+		})
+
+		ginkgo.It(fmt.Sprintf("%s set", sysctlVar), func() {
+			ginkgo.By("getting a list of worker nodes")
+			nodes, err := util.GetNodesByRole(cs, "worker")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
is there something we can do with this gomega prefix while keeping the dot imports guidelines ? its a built bulky.

Reply:
If you look upstream k8s or OpenShift tests, none of them use dot imports as dot imports are considered bad practice.  Also, I'm concerned about a possible removal of dot imports in future golang 2.x: https://github.com/golang/go/issues/29326